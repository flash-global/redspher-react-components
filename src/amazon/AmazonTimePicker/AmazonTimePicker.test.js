import React from 'react'
import renderer from 'react-test-renderer'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { AmazonTimePicker } from '../../index'

configure({ adapter: new Adapter() })

const testProps = {
  ampm: false,
  disabled: false,
  value: '00:00',
  helperText: 'time is invalid',
  emptyLabel: 'no time',
  onClick: jest.fn(),
  onChange: jest.fn()
}

/** Snapshots */
it('renders correctly props object', () => {
  const tree = renderer.create(<AmazonTimePicker {...testProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('has a value', () => {
  const tree = renderer
    .create(
      <AmazonTimePicker
        value={testProps.value}
        onClick={testProps.onClick}
        onChange={testProps.onChange}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('is on 24h', () => {
  const tree = renderer
    .create(
      <AmazonTimePicker
        ampm
        value={testProps.value}
        onClick={testProps.onClick}
        onChange={testProps.onChange}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('is disabled', () => {
  const tree = renderer
    .create(
      <AmazonTimePicker
        disabled
        value={testProps.value}
        onClick={testProps.onClick}
        onChange={testProps.onChange}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
