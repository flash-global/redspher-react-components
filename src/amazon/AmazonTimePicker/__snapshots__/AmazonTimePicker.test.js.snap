// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a value 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root defaultTimePicker"
  onClick={[MockFunction]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  
</div>
`;

exports[`is disabled 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root defaultTimePicker"
  onClick={[MockFunction]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd"
      disabled={true}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  
</div>
`;

exports[`is on 24h 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root defaultTimePicker"
  onClick={[MockFunction]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  
</div>
`;

exports[`renders correctly props object 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root defaultTimePicker"
  onClick={[MockFunction]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <p
    className="MuiFormHelperText-root Mui-error"
  >
    time is invalid
  </p>
</div>
`;
